src/components/common/Map.js

```diff
@@ -1,26 +1,59 @@
-import React, {PropTypes} from 'react';
-import {GoogleMapLoader, GoogleMap, Marker} from 'react-google-maps';
+import React, { PropTypes } from 'react';
+import { GoogleMapLoader, GoogleMap, Marker } from 'react-google-maps';
 
 class Map extends React.Component {
-  constructor(props, context){
+  constructor(props, context) {
     super(props, context);
+
+    this.onMapClick = this.onMapClick.bind(this);
   }
 
-  render() {
-    const mapContainer = <div style={{position: 'absolute', height:'100%', width:'100%'}}></div>;
+  onMapClick(e) {
+    let lat = parseFloat(e.latLng.lat().toFixed(6));
+    let lng = parseFloat(e.latLng.lng().toFixed(6));
+    let marker = {
+      lat: lat,
+      lng: lng,
+      key: 2,
+      defaultAnimation: 2,
+    };
+    this.props.addMarker(marker);
+  }
 
-    return(
-      <GoogleMapLoader
-      containerElement = {mapContainer}
-      googleMapElement = {
-        <GoogleMap
-          defaultZoom = {15}
-          defaultCenter={{ lat: 46.2017559, lng: 6.1466014 }}
-          options={{streetViewControl: false, mapTypeControl: false}}>
-        </GoogleMap>
-      } />
+  render() {
+    const mapContainer = <div style={{ position: 'absolute', height: '100%', width: '100%' }} />;
```

Comment:
Try not to use inline styling, CSS when possible, else declare a const variable somewhere near and use it here.